File: chdsr/chdsr-simulator-web/src/main/java/io/iochord/dev/chdsr/simulator/web/v1/models/WebServiceStatus.java
package io.iochord.dev.chdsr.simulator.web.v1.models;

import java.util.ArrayList;
import java.util.List;

import lombok.Getter;
import lombok.Setter;

/**
 *
 * @package chdsr-simulator-web
 * @author  Iq Reviessay Pulshashi <pulshashi@ideas.web.id>
 * @since   2019
 *
 *
 */
public class WebServiceStatus {
	
	public static WebServiceStatus completed() {
		return new WebServiceStatus("completed");
	}
	
	public static  WebServiceStatus submitted() {
		return new WebServiceStatus("submitted");
	}

	@Getter
	private final List<WebServiceError> errors = new ArrayList<>();

	@Getter
	private final String status;

	@Getter
	@Setter
	private String progressWsUri;

	@Getter
	@Setter
	private String completeWsUri;
	
	public WebServiceStatus(String status) {
		this.status = status;
	}
	
}

File: ips/ips-simulator-web/src/main/ts/src/iochord/ips/common/service/WebServiceStatus.ts
/**
 * The web socket status class.
 *
 * @export
 * @class WebServiceStatus
 *
 * @package ips
 * @author  Iq Reviessay Pulshashi <pulshashi@ideas.web.id>
 * @since   2019
 *
 */
export class WebServiceStatus {

  /**
   * Errors object retreive from web socket.
   *
   * @type {*}
   * @memberof WebServiceStatus
   */
  public errors: any = null;

  /**
   * Status of web socket.
   *
   * @type {string}
   * @memberof WebServiceStatus
   */
  public status: string = '';

  /**
   * Progress of the web socket.
   *
   * @type {string}
   * @memberof WebServiceStatus
   */
  public progressWsUri: string = '';

  /**
   * Response from the web socket after complete listening.
   *
   * @type {string}
   * @memberof WebServiceStatus
   */
  public completeWsUri: string = '';
}
